optimize/backend/src/main/java/io/camunda/optimize/tomcat/SaasRequestAdjustmentFilter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.tomcat;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.RequestDispatcher;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class SaasRequestAdjustmentFilter implements Filter {
```

Comment:
Some code in this filter looks similar to the code in `CcsmRequestAdjustmentFilter`, could you refactor the common code so that nothing is duplicated?

Reply:
When I tried the refactoring, I saw that the duplicate part is this:
```
if (requestURI.startsWith("/external/api/")) {
      final String rewrittenURI = requestURI.replaceFirst("/external/api/", "/api/external/");
```
everything else has small differences (like the `forward = true` that is only done in 1 class) that would cost more overhead with the refactoring.

Although I don't have a strong opinion about this: if it's a blocker for you, I'll try to refactor it, otherwise I'd prefer leaving it as it is now, for simplicity